containers/src/Kokkos_DynRankView.hpp

```diff
@@ -1089,19 +1089,8 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
           "execution space");
     }
 
-//------------------------------------------------------------
-#if defined(KOKKOS_ENABLE_CUDA)
-    // If allocating in CudaUVMSpace must fence before and after
-    // the allocation to protect against possible concurrent access
-    // on the CPU and the GPU.
-    // Fence using the trait's executon space (which will be Kokkos::Cuda)
-    // to avoid incomplete type errors from usng Kokkos::Cuda directly.
-    if (std::is_same<Kokkos::CudaUVMSpace,
-                     typename traits::device_type::memory_space>::value) {
-      typename traits::device_type::memory_space::execution_space().fence(
-          "Kokkos::DynRankView<>::DynRankView: fence before UVM allocation");
-    }
-#endif
+    //------------------------------------------------------------
+    Impl::allocation_fence_if_shared_space<traits>(arg_prop, "Kokkos::DynRankView<>::DynRankView: fence before UVM allocation");
     //------------------------------------------------------------
```

Comment:
I don't think this fence is necessary. We can't access the new allocation before it was created.

Reply:
OK, so let's remove the 3 "fence before allocation" (`View`, `OffsetView` and `DynRankView`) :person_fencing:

Reply:
I'm closing this thread, because it was decided not to fence before allocation (https://github.com/kokkos/kokkos/pull/6008#discussion_r1146208102).